Add Repo Layout Migration doc (#159)
osg-htc/technology: docs/policy/repo-layout-migration-2025.md

OSG Software Repo Layout Migration
==================================

11 December 2024

> On Monday, January 6th, 2025, the OSG Software Team will be upgrading the server that hosts the OSG Software repos

Reviewer: ```suggestion
On Monday, January 6th, 2025, the OSG Software Team will be upgrading the server that hosts the OSG Software Yum repositories
```

> (<https://repo.osg-htc.org> a.k.a. <https://repo.opensciencegrid.org>),

Reviewer: ```suggestion
(<https://repo.osg-htc.org> nee <https://repo.opensciencegrid.org>),
```

Or née if we want our pinkies up

> which will result in changes to the directory layout for the OSG 23 and OSG 24 YUM repos.

Reviewer: ```suggestion
which will result in changes to the directory layout for the OSG 23 and OSG 24 Yum repos.
```

RedHat is all over the place on capitalization here but we should try to stay on the same page. I think we capitalize it elsewhere.

> Users should not be affected since the URLs of the repos themselves will not change.

Reviewer: ```suggestion
Users installing packages should not be affected since the URLs of the repositories themselves will not change.
```

We have so many different kinds of users

> However, mirrors will have to take some action to avoid rsyncing those repos from scratch.

Reviewer: ```suggestion
However, mirror administrators will have to take some action to avoid rsyncing those repos from scratch.
```

AI has gone too far if the mirror services are making the adjustments themselves

Perform the following steps:

> 1.  Turn off rsync from `repo-rsync.opensciencegrid.org` (a.k.a. `repo-rsync.osg-htc.org`)

Reviewer: ```suggestion
1.  Turn off rsync from `repo-rsync.opensciencegrid.org` or `repo-rsync.osg-htc.org`
```

Maybe?

2.  Download the migration script from <https://github.com/osg-htc/osg-repo-scripts/blob/el9/migrate.py>

>     ([direct link](https://raw.githubusercontent.com/osg-htc/osg-repo-scripts/refs/heads/el9/migrate.py))

Reviewer: Do we want to provide a checksum or something?

Author: I don't think anybody will verify them on a file this small.

> 3.  Run the migration script on your OSG 23 and OSG 24 repos.

Reviewer: ```suggestion
3.  Run the migration script on your OSG 23 and OSG 24 Yum repositories.
```

    For example, if the files on your mirror are under `/mnt/mirror/osg`, then run

        :::shell
        python3 migrate.py /mnt/mirror/osg/23-* /mnt/mirror/osg/24-*

4.  Switch the rsync source to repo-rsync-itb.osg-htc.org and add `--delay-updates --delete-delay` to the rsync command.
    For example, if your rsync command is typically

        :::shell
        rsync -av --delete \
            repo-rsync.opensciencegrid.org::osg/ \
            /mnt/mirror/osg

    then change it to

        :::shell
        rsync -av --delay-updates --delete-delay \
            repo-rsync-itb.osg-htc.org::osg/ \
            /mnt/mirror/osg

This will switch your mirror to the new repo layout, minimizing disruption to users.

On January 6th, we will upgrade <https://repo.osg-htc.org>, at which point you should change your rsync source

> from `repo-rsync-itb.osg-htc.org` to `repo-rsync.osg-htc.org`

Reviewer: ```suggestion
from `repo-rsync-itb.osg-htc.org` to `repo-rsync.osg-htc.org`.
```

Maybe worth adding a note that we will send a reminder about this post update
